app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,31 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
+    @posts = Post.where(author_id: @user)
   end
 
   def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
```

Comment:
- [**Optional**] Hi @Nowembabazi we can equally achieve this using a single line of code. Kindly check it out below if interested 😄 

`  @post = current_user.posts.new(post_params)`